xclim/core/calendar.py

```diff
@@ -340,7 +340,7 @@ def _convert_datetime(
     datetime: Union[pydt.datetime, cftime.datetime],
     new_doy: Optional[Union[float, int]] = None,
     calendar: str = "default",
-) -> Union[cftime.datetime, pydt.datetime, np.int]:
+) -> Union[cftime.datetime, pydt.datetime, int]:
```

Comment:
I addressed this in #716 (and np.nan is technically a float!)

Reply:
Oh woups. I'll revert this. Thanks!